core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -129,6 +132,27 @@ public TableMetadata refresh() {
 
   @Override
   public void commit(TableMetadata base, TableMetadata metadata) {
+    // dropOldVersionHint----------->JobFail
```

Comment:
Please remove the diagram, if you need to document the logic I would add it to the Javadoc but I don't think that's critical here either

Reply:
Okay